src/components/gltf-model-plus.js

```diff
@@ -325,7 +324,15 @@ AFRAME.registerComponent("gltf-model-plus", {
         return;
       }
 
+      window.stuffToLoad += 1;
+      let text = `Loading ${window.stuffToLoad} more object${window.stuffToLoad > 1 ? "s" : ""}...`;
+      window.uiroot && window.uiroot.setState({ loadingText: text });
       const gltf = await this.loadModel(src, contentType, this.preferredTechnique, this.data.useCache);
+      window.stuffToLoad -= 1;
+      if (window.stuffToLoad > 0) {
+        text = `Loading ${window.stuffToLoad} more object${window.stuffToLoad > 1 ? "s" : ""}...`;
+        window.uiroot && window.uiroot.setState({ loadingText: text });
+      }
```

Comment:
Hmm, this code is pretty surprising to me.
Firstly, we should probably not just treat `window` like a grab-bag of properties. If we do have to use global state like this, we should at least namespace it under our existing `APP` global, maybe under an `APP.resourceLoading` namespace as well.

It's also surprising to have an a-frame component call `setState` directly on a react component, also via property on the window global

Reply:
I think a less surprising way of doing this is to emit `loading-started` and `loading-ended` events, and then handle the incrementing/decrementing and UI updating in the react code instead, or are you really trying to avoid using events?